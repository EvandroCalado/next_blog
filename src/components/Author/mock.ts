export default {
  title: 'Evandro Calado',
  resume: 'Desenvolvedor Frontend',
  image: 'https://avatars.githubusercontent.com/u/110628201?v=4',
  alt: 'Evandro Calado',
};
