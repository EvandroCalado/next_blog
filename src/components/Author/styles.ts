import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 3rem;
  margin: ${theme.spacings.xxlarge} 0;
`}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 15rem;
    border: 3px solid ${theme.colors.secondary};
    border-radius: 50%;
  `}
`;

export const Container = styled.div`
`;

export const Title = styled.h2`
${({ theme }) => css`
font-size: ${theme.font.sizes.xlarge};
`}
`;

export const Paragraph = styled.p`
`;
