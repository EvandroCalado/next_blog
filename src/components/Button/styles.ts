import styled, { css } from 'styled-components';
import { ButtonProps } from './';

export const Button = styled.button<Pick<ButtonProps, 'color'>>`
  ${({ theme, color }) => css`
    background-color: ${theme.colors[color]};
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.medium};
    border: none;
    padding: ${theme.spacings.xsmall} ${theme.spacings.medium};
    margin: ${theme.spacings.large} 0;
    border-radius: 0.5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all 300ms ease-in-out;
    cursor: pointer;

    &:hover {
      filter: brightness(80%);
    }

    &:disabled {
      background-color: ${theme.colors.mediumGray};
      color: ${theme.colors.darkGray};
      cursor: not-allowed;

      &:hover {
        filter: none;
      }
    }

    > svg {
      width: 3rem;
      margin-left: 1rem;
    }
  `}
`;
