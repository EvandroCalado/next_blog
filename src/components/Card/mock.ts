export default {
  slug: 'slug-test',
  title: 'Title',
  cover:
    'https://images.unsplash.com/photo-1593720219276-0b1eacd0aef4?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=640&q=80',
  excerpt:
    'JavaScript é uma linguagem de programação popular e amplamente utilizada para criar e desenvolver aplicativos web dinâmicos e interativos. Ela é interpretada pelo navegador do usuário e é executada em tempo real, permitindo que os desenvolvedores adicionem funcionalidades avançadas aos sites, como animações, interatividade do usuário, validação de formulários, solicitações assíncronas ao servidor e muito mais.',
};
