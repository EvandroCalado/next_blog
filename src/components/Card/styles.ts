import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    a {
      transition: all 300ms ease-in-out;
    }

    a:hover {
      opacity: 1;
      filter: brightness(170%);
    }
`;

export const Cover = styled.div`
  ${({ theme }) => css`
    img {
      width: 100%;
      display: block;
      aspect-ratio: 16/9;
      border-radius: .5rem;
      margin-bottom: ${theme.spacings.small};

    }
  `}
`;

export const Heading = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.smallMedium};
    color: ${theme.colors.darkGray};
    margin-bottom: ${theme.spacings.xsmall};
  `}
`;

export const Paragraph = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    margin-bottom: ${theme.spacings.small};
    color: ${theme.colors.primary};
  `}
`;
