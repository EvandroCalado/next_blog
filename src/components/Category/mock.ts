export default {
  categories: [
    {
      image:
        'https://res.cloudinary.com/dnhsdenji/image/upload/v1682864831/javascript_programming_language_icon_ce8f9d2d57.png',
      title: 'Javascript',
    },
    {
      image:
        'https://res.cloudinary.com/dnhsdenji/image/upload/v1682864831/javascript_programming_language_icon_ce8f9d2d57.png',
      title: 'Javascript',
    },
  ],
  category: 'Javascript',
};
