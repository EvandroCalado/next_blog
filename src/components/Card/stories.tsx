import PostCard, { PostCardProps } from '.';
import { Meta, StoryFn } from '@storybook/react';

import mock from './mock';

export default {
  title: 'PostCard',
  component: PostCard,
  args: mock,
} as Meta;

export const Template: StoryFn<PostCardProps> = (args) => (
  <div>
    <PostCard {...args} />
  </div>
);
