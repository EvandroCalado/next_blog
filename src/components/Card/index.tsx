import Link from 'next/link';
import * as Styled from './styles';
import Date from '../Date';

export type CardProps = {
  slug: string;
  title: string;
  cover: string;
  createdAt: string;
};

const Card = ({ slug, title, cover, createdAt }: CardProps) => {
  return (
    <Styled.Wrapper>
      <Link href={`/post/${slug}`}>
        <Styled.Cover>
          <img src={cover} alt={title} />
        </Styled.Cover>
        <Styled.Heading>{title}</Styled.Heading>
        <Styled.Paragraph>
          <Date date={createdAt} />
        </Styled.Paragraph>
      </Link>
    </Styled.Wrapper>
  );
};

export default Card;
