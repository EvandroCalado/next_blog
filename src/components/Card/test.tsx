import Card from './';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';

import mock from './mock';

describe('<Card />', () => {
  it('should render a card with values', () => {
    const { container } = renderTheme(<Card {...mock} />);

    expect(screen.getByText(mock.title)).toBeInTheDocument();
    expect(screen.getByText('30/04/2023')).toBeInTheDocument();
    expect(screen.getByRole('link')).toHaveAttribute(
      'href',
      `/post/${mock.slug}`,
    );
    expect(container).toMatchSnapshot();
  });
});
