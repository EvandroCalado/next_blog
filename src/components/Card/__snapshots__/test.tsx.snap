// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render a card with values 1`] = `
.c0 a {
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  opacity: 1;
  -webkit-filter: brightness(170%);
  filter: brightness(170%);
}

.c1 img {
  width: 100%;
  display: block;
  aspect-ratio: 16/9;
  border-radius: .5rem;
  margin-bottom: 1.6rem;
}

.c2 {
  font-size: 2rem;
  color: #333333;
  margin-bottom: 0.8rem;
}

.c3 {
  font-size: 1.6rem;
  margin-bottom: 1.6rem;
  color: #252422;
}

.c4 {
  color: #f24b4b;
  font-style: italic;
  margin-right: 1rem;
}

<div>
  <div
    class="c0"
  >
    <a
      href="/post/slug-test"
    >
      <div
        class="c1"
      >
        <img
          alt="Title"
          src="https://images.unsplash.com/photo-1593720219276-0b1eacd0aef4?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=640&q=80"
        />
      </div>
      <h2
        class="c2"
      >
        Title
      </h2>
      <p
        class="c3"
      >
        <span
          class="c4"
        >
          30/04/2023
        </span>
      </p>
    </a>
  </div>
</div>
`;
